platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ApplicationDefinitionLoader.java

```diff
@@ -93,9 +93,11 @@ private ApplicationDefinitionLoader() {}
         // the line is: app, jarFilename, optionalParameters
         final String appJarFilename = appConfig.jarName();
         // this is a real .jar file, so load from data/apps/
-        Path appJarPath = ConfigurationHolder.getConfigData(PathsConfig.class)
-                .getAppsDirPath()
-                .resolve(appJarFilename);
+        final PathsConfig defaultPathsConfig = ConfigurationBuilder.create()
```

Comment:
Can we pass the config as a param to the method. Since this is not a main method it can end in strange behavior if somebody assumes that settings.txt is used.

Reply:
Done.